// recipe-ingredient-parser-v3/testDist/src/units.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unitsMap = exports.itaPreposition = exports.itaNameToSymbol = exports.itaPluralUnits = exports.itaUnits = exports.engPreposition = exports.engNameToSymbol = exports.engPluralUnits = exports.engUnits = void 0;
exports.engUnits = {
    bag: ['bag', 'bags'],
    box: ['box'],
    can: ['can'],
    cup: ['cup', 'c', 'c.', 'C', 'Cups'],
    clove: ['clove'],
    gallon: ['gallon', 'gal'],
    ounce: ['ounce', 'oz', 'oz.'],
    pint: ['pint', 'pt', 'pts', 'pt.'],
    pound: ['pound', 'lb', 'lb.', 'lbs', 'lbs.', 'Lb', 'Lbs'],
    quart: ['quart', 'qt', 'qt.', 'qts', 'qts.'],
    tablespoon: ['tbs', 'tbsp', 'tbspn', 'T', 'T.', 'Tablespoons', 'Tablespoon'],
    teaspoon: ['teaspoon', 'tsp', 'tspn', 't', 't.'],
    gram: ['gram', 'g', 'g.'],
    kilogram: ['kilogram', 'kg', 'kg.', 'Kg', 'Kg.'],
    liter: ['liter', 'l', 'l.', 'lt', 'Lt', 'LT', 'L', 'L.'],
    milligram: ['milligram', 'mg', 'mg.'],
    milliliter: ['milliliter', 'ml', 'ml.', 'mL', 'mL.'],
    package: ['package', 'pkg', 'pkgs'],
    stick: ['stick', 'sticks'],
    piece: ['piece', 'pcs', 'pcs.'],
    pinch: ['pinch'],
    small: ['Small'],
    slice: ['slice'],
    medium: ['Medium'],
    large: ['large', 'Large'],
};
exports.engPluralUnits = {
    cup: 'cups',
    gallon: 'gallons',
    ounce: 'ounces',
    pint: 'pints',
    pound: 'pounds',
    quart: 'quarts',
    tablespoon: 'tablespoons',
    teaspoon: 'teaspoons',
    gram: 'grams',
    kilogram: 'kilograms',
    liter: 'liters',
    milligram: 'milligrams',
    milliliter: 'milliliters',
    clove: 'cloves',
    bag: 'bags',
    box: 'boxes',
    pinch: 'pinches',
    can: 'cans',
    slice: 'slices',
    piece: 'pieces'
};
exports.engNameToSymbol = {
    cup: 'c',
    gallon: 'gal',
    ounce: 'oz',
    pint: 'pt',
    pound: 'lb',
    quart: 'qt',
    tablespoon: 'tbs',
    teaspoon: 'tsp',
    gram: 'g',
    kilogram: 'kg',
    liter: 'lt',
    milligram: 'mg',
    milliliter: 'ml',
    clove: '',
    bag: '',
    box: '',
    pinch: '',
    can: '',
    slice: '',
    piece: ''
};
exports.engPreposition = ['of'];
exports.itaUnits = {
    bicchiere: ['bicchiere'],
    bustina: ['bustina', 'bustine'],
    cucchiaio: ['cucchiai', 'cucchiaio'],
    cucchiaino: ['cucchiaini', 'cucchiaino'],
    grammo: ['g', 'g\\.', 'gr', 'gr\\.', 'grammi', 'grammo'],
    chilogrammo: ['kg.', 'kg', 'kilogrammo', 'chilogrammi', 'kilogrammo', 'chilogrammo'],
    fetta: ['fetta', 'fette'],
    foglia: ['foglie', 'foglia'],
    litro: ['l\\.', 'l', 'lt', 'litro'],
    mazzetto: ['Mazzetto', 'mazzetti', 'mazzetto'],
    lattina: ['Lattina', 'lattina'],
    milligrammo: ['mg.', 'mg', 'milligrammo'],
    millilitro: ['ml', 'ml\\.', 'millilitro'],
    panetto: ['Panetto', 'panetti', 'panetto'],
    pacco: ['pkg', 'pkgs', 'pacchetto', 'pacco'],
    pezzo: ['pezzo', 'pcs', 'pcs.', 'pezzi'],
    pizzico: ['pizzico', 'pizzichi'],
    tazza: ['tazza', 'tazzina', 'tazzine'],
    spicchio: ['spicchio', 'spicchi'],
    scatola: ['scatola', 'scatole'],
    vasetto: ['vasetto', 'vasetti'],
    filo: ['filo'],
    ciuffo: ['ciuffo'],
    scatoletta: ['scatoletta'],
    manciata: ['manciata'],
    rotolo: ['rotolo'],
    bicchierino: ['bicchierino'],
    //noce: ['noce'],
};
exports.itaPluralUnits = {
    bicchiere: 'bicchieri',
    bustina: 'bustine',
    tazza: 'tazze',
    quarto: 'quarti',
    cucchiaio: 'cucchiai',
    cucchiaino: 'cucchiaini',
    grammo: 'grammi',
    chilogrammo: 'chilogrammi',
    litro: 'litri',
    milligrammo: 'milligrammi',
    millilitro: 'millilitri',
    spicchio: 'spicchi',
    scatola: 'scatole',
    pizzico: 'pizzichi',
    lattina: 'lattine',
    fetta: 'fette',
    pezzo: 'pezzi',
    panetto: 'panetti',
    foglia: 'foglie',
    mazzetto: 'mazzetti',
    vasetto: 'vasetti',
    filo: 'fili',
    ciuffo: 'ciuffi',
    scatoletta: 'scatolette',
    manciata: 'manciate',
    rotolo: 'rotoli',
    bicchierino: 'bicchierini',
    //noce: 'noci'
};
exports.itaNameToSymbol = {
    bicchiere: '',
    bustina: '',
    tazza: '',
    quarto: '',
    cucchiaio: '',
    spicchio: '',
    scatola: '',
    pizzico: '',
    lattina: '',
    fetta: '',
    pezzo: '',
    panetto: '',
    foglia: '',
    mazzetto: '',
    vasetto: '',
    grammo: 'g',
    cucchiaino: 'cc',
    chilogrammo: 'kg',
    litro: 'lt',
    milligrammo: 'mg',
    millilitro: 'ml',
};
exports.itaPreposition = ['di', 'd\''];
exports.unitsMap = new Map();
exports.unitsMap.set("eng", [exports.engUnits, exports.engPluralUnits, exports.engPreposition, exports.engNameToSymbol]);
exports.unitsMap.set("ita", [exports.itaUnits, exports.itaPluralUnits, exports.itaPreposition, exports.itaNameToSymbol]);
